contracts/PoaOperator.sol

```diff
@@ -84,8 +84,20 @@ contract PoaOperator is Adminable {
     emit EpochLength(epochLength);
   }
 
+  function _isEmpty(Slot memory _slot) internal returns (bool) {
+    return (_slot.signer == address(0));
+  }
+
+  function _getLargestSlot() internal returns (uint256) {
+    uint256 slotId = epochLength;
+    do {
+      slotId--;
+    } while (_isEmpty(slots[slotId]));
```

Comment:
I thought of doing this, but immediately dismissed with "that's extra gas and Johann is reviewer" 😃

Reply:
it is hard to know really. but I found that simple computation and internal function almost don't affect the gas cost.